### ISSUE TITLE
[BUG] [Java] [spring-boot] Code generated from a specification with a default response indicates the wrong return code

### ISSUE THREAD

**Issue by GregDThomas** [OpenAPITools/openapi-generator#14394]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description
The specification allows a "default" response to be used, typically for error payloads - https://swagger.io/specification/#path-item-object-example

If code is generated using a "default" response then the wrong HTTP return code is indicated;
```java
    @Operation(
        operationId = "getPetsById",
        summary = "Find pets by ID",
        responses = {
            @ApiResponse(responseCode = "200", description = "pet response", content = {
                @Content(mediaType = "application/xml", array = @ArraySchema(schema = @Schema(implementation = Pet.class))),
                @Content(mediaType = "application/json", array = @ArraySchema(schema = @Schema(implementation = Pet.class)))
            }),
            @ApiResponse(responseCode = "200", description = "error payload"
            )
        }
    )
```
Note the responseCode for the "error payload" is 200. It should be "default".

This means anyone examining the generated code will get a misleading idea of what the response could be.

##### openapi-generator version
6.2.1

**Comment by borsch** [comment]
@GregDThomas please try latest 6.3.0-SNAPSHOT

**Comment by GregDThomas** [comment]
This all looks good. Thank you.

**Comment by martinznamenacek-ext91575** [comment]
Hi, exactly the same issue is in kotlin-spring in version 7.14.0. Fix should be very easy as was in #14399.
